tests/core/state.cpp

```diff
@@ -643,16 +646,19 @@ namespace TState {
       P2P::ManagerDiscovery p2pDiscovery(
           boost::asio::ip::address::from_string("127.0.0.1"), discoveryOptions);
 
-      // References for the rdPoS workers vector.
-      std::vector<std::reference_wrapper<rdPoS>> rdPoSreferences;
-      rdPoSreferences.emplace_back(blockchainWrapper1.rdpos);
-      rdPoSreferences.emplace_back(blockchainWrapper2.rdpos);
-      rdPoSreferences.emplace_back(blockchainWrapper3.rdpos);
-      rdPoSreferences.emplace_back(blockchainWrapper4.rdpos);
-      rdPoSreferences.emplace_back(blockchainWrapper5.rdpos);
-      rdPoSreferences.emplace_back(blockchainWrapper6.rdpos);
-      rdPoSreferences.emplace_back(blockchainWrapper7.rdpos);
-      rdPoSreferences.emplace_back(blockchainWrapper8.rdpos);
+      // Vector of references for the states' rdPoS workers
+      // (rdPoS is exclusively owned by State and can't be exposed in any way,
```

Comment:
If their lifetimes are bound together (rdPoS is a reference to a part of State) and you don't need to add e.g. extra synchronization for the access, then perhaps it is fine to just allow code to get a reference to it?

Reply:
Same as the previous comments.